roles/ycsb-bench/tasks/main.yml

```diff
@@ -1,98 +1,127 @@
+                command: ["/bin/sh"]
+                args: ["-c", "/ycsb/bin/ycsb load {{ ycsb.driver }} -s -P /ycsb/workloads/{{ ycsb.workload }} {{ ycsb_flags }}"]
+              restartPolicy: Never
+          backoffLimit: 4
+
+  - name: Wait for Load Job to Succeed...
+    k8s_facts:
+      kind: Job
+      api_version: batch/v1
+      name: '{{ meta.name }}-ycsb-data-load-job'
+      namespace: "{{ operator_namespace }}"
+    register: ycsb_load_pod
+    until: "ycsb_load_pod | json_query('resources[].status.succeeded')"
+    retries: 30
+    delay: 10
 
-#- name: Clean up Data Load Pod
-#  k8s:
-#    state: absent
-#    definition:
-#      kind: Pod
-#      apiVersion: v1
-#      metadata:
-#        name: '{{ meta.name }}-ycsb-data-load'
-#        namespace: '{{ meta.namespace }}'
+  - name: Run YCSB Workload
+    k8s:
+      definition:
+        kind: Job
+        apiVersion: batch/v1
+        metadata:
+          name: '{{ meta.name }}-ycsb-bench-job'
+          namespace: '{{ operator_namespace }}'
+        spec:
+          template:
+            metadata:
+              name: '{{ meta.name }}-ycsb-bench'
+              namespace: '{{ operator_namespace }}'
+              labels:
+                name: 'ycsb-run'
+            spec:
+              containers:
+              - name: ycsb
+                image: "quay.io/cloud-bulldozer/ycsb-server:latest"
+                imagePullPolicy: Always
+                command: ["/bin/sh"]
+                args: ["-c", "/ycsb/bin/ycsb run {{ ycsb.driver }} -s -P /ycsb/workloads/{{ ycsb.workload }} {{ ycsb_flags }}"]
+              restartPolicy: Never
+          backoffLimit: 4
 
-- name: Run YCSB Workload
-  k8s:
-    definition:
+  - name: Wait for YCSB Workload Job to Succeed...
+    k8s_facts:
       kind: Job
-      apiVersion: batch/v1
-      metadata:
-        name: '{{ meta.name }}-ycsb-bench-job'
-        namespace: '{{ operator_namespace }}'
-      spec:
-        template:
-          metadata:
-            name: '{{ meta.name }}-ycsb-bench'
-            namespace: '{{ operator_namespace }}'
-            labels:
-              name: 'ycsb-run'
-          spec:
-            containers:
-            - name: ycsb
-              image: "quay.io/cloud-bulldozer/ycsb-server:latest"
-              imagePullPolicy: Always
-              command: ["/bin/sh"]
-              args: ["-c", "/ycsb/bin/ycsb run {{ ycsb.driver }} -s -P /ycsb/workloads/{{ ycsb.workload }} {{ ycsb_flags }}"]
-            restartPolicy: Never
-        backoffLimit: 4
+      api_version: batch/v1
+      name: '{{ meta.name }}-ycsb-bench-job'
+      namespace: "{{ operator_namespace }}"
+    register: ycsb_bench
+    until: "ycsb_bench | json_query('resources[].status.succeeded')"
+    retries: 60
+    delay: 10
+
+
+# Destroy block is run when the finalizer has set teardown=true
+- name: Destroy
+  when: teardown is defined
+  block:
+
+  - name: Get YCSB workload pod
+    k8s_facts:
+      kind: Pod
```

Comment:
This should be a job, and wouldn't operator-framework by default cleanup resources the operaator created which in this case is the jobs